src/audio/volume/volume.h

```diff
@@ -92,16 +112,22 @@ struct sof_ipc_ctrl_value_chan;
  */
 #define PEAK_16S_32C_ADJUST 16
 
-/**
- * \brief Volume maximum value.
- * TODO: This should be 1 << (VOL_QX_BITS + VOL_QY_BITS - 1) - 1 but
- * the current volume code cannot handle the full Q1.16 range correctly.
- */
+#if COMP_VOLUME_Q1_31
+/** \brief Volume maximum value. */
+#define VOL_MAX		INT32_MAX
+
+/** \brief Volume 0dB value. */
+#define VOL_ZERO_DB	INT32_MAX
```

Comment:
Maybe it's correct and maybe it's just me, but it does look a bit confusing, or at least unclear. Maybe at least a bit larger comment could help. Is this the same value because by convention volume doesn't get positive or do these two macros have different usage?

Reply:
In IPC4 due to Q1.31 format the max volume and (near) 0 dB gain value are the same since the max linear gain is 0.9999... In IPC3 we used Q-formats those allowed amplification in volume component. I don't want to yet remove the IPC3 gain range since we still have IPC3 in the code. But we can do it later if we remove IPC3 from git main. But I don't think the Q1.31 is a good choice since it can't represent exactly gain 1 and it can't amplify. But that's what we have now.